synapse/http/site.py

```diff
@@ -368,7 +381,10 @@ def _finished_processing(self):
 
         # the time between the request handler finishing and the response being sent
         # to the client (nb may be negative)
-        response_send_time = self.finish_time - self._processing_finished_time
+        # TODO can we avoid the cast? Maybe take finish_time as an explicit float param?
+        response_send_time = (
+            cast(float, self.finish_time) - self._processing_finished_time
```

Comment:
I think this is slightly less evil.

```suggestion
        assert self.finish_time is not None
        response_send_time = (
            self.finish_time - self._processing_finished_time
```